src/game/Laptop/IMP_SkillTraits.cc

```diff
@@ -0,0 +1,687 @@
+			{
+			}
+			else
+			{
+				//We are finished on this page
+				iCurrentProfileMode = 2;
+			}
+		}
+	}
+}
+
+
+BOOLEAN CameBackToSpecialtiesPageButNotFinished()
+{
+	//if we are in a page that comes after this one
+	if( iCurrentProfileMode == 2 )
+	{
+		return( TRUE );
+	}
+	else
+	{
+		return( FALSE );
+	}
+}
+
+INT8	DoesPlayerHaveExtraAttibutePointsToDistributeBasedOnSkillSelection()
+{
+	INT8		bExtraPoints=10;
+	INT8		bNumSkills=0;
+
+	//Count the number of skills selected
+	bNumSkills = CountNumSkillStraitsSelected( FALSE );
+
+	//if there NONE selected
+	if( bNumSkills == 0 )
+	{
+		bExtraPoints = 10;
+	}
+
+	//if there is 1 trait selected
+	else if( bNumSkills == 1 )
+	{
+		//if the trait is not the ones you cant be expert in )
+		if( gfSkillTraitQuestions[ IMP_SKILL_TRAITS__ELECTRONICS ] ||
+				gfSkillTraitQuestions[ IMP_SKILL_TRAITS__AMBIDEXTROUS  ] ||
+				gfSkillTraitQuestions[ IMP_SKILL_TRAITS__CAMO ] )
+		{
+			bExtraPoints = 5;
+		}
+		else
+		{
+			bExtraPoints = 0;
+		}
+	}
+
+	//else both traits selected
+	else
+	{
+		bExtraPoints = 0;
+	}
+
+	return( bExtraPoints );
+}
+
+
+
+BOOLEAN ShouldTraitBeSkipped( UINT32 uiTrait )
+{
+	if( uiTrait == IMP_SKILL_TRAITS__MARTIAL_ARTS && !fCharacterIsMale )
+		return( TRUE );
+	else
+		return( FALSE );
+}
+
+
+void AddSelectedSkillsToSkillsList()
+{
+	UINT32	uiCnt;
+
+	//loop through all the buttons and reset them
+	for( uiCnt=0; uiCnt<IMP_SKILL_TRAITS__NONE; uiCnt++ )
+	{
+		//if the trait is selected
+		if( gfSkillTraitQuestions[ uiCnt ] )
+		{
+			//switch on the trait, and add it to the list
+			switch( uiCnt )
+			{
+				case IMP_SKILL_TRAITS__LOCKPICK:
+					AddSkillToSkillList( LOCKPICKING );
+					break;
+
+				case IMP_SKILL_TRAITS__HAND_2_HAND:
+					AddSkillToSkillList( HANDTOHAND );
+					break;
+
+				case IMP_SKILL_TRAITS__ELECTRONICS:
+					AddSkillToSkillList( ELECTRONICS );
+					break;
+				
+				case IMP_SKILL_TRAITS__NIGHT_OPS:
+					AddSkillToSkillList( NIGHTOPS );
+					break;
+				
+				case IMP_SKILL_TRAITS__THROWING:
+					AddSkillToSkillList( THROWING );
+					break;
+				
+				case IMP_SKILL_TRAITS__TEACHING:
+					AddSkillToSkillList( TEACHING );
+					break;
+				
+				case IMP_SKILL_TRAITS__HEAVY_WEAPONS:
+					AddSkillToSkillList( HEAVY_WEAPS );
+					break;
+				
+				case IMP_SKILL_TRAITS__AUTO_WEAPONS:
+					AddSkillToSkillList( AUTO_WEAPS );
+					break;
+				
+				case IMP_SKILL_TRAITS__STEALTHY:
+					AddSkillToSkillList( STEALTHY );
+					break;
+				
+				case IMP_SKILL_TRAITS__AMBIDEXTROUS:
+					AddSkillToSkillList( AMBIDEXT );
+					break;
+				
+				case IMP_SKILL_TRAITS__KNIFING:
+					AddSkillToSkillList( KNIFING );
+					break;
+				
+				case IMP_SKILL_TRAITS__ROOFTOP_SNIPING:
+					AddSkillToSkillList( ONROOF );
+					break;
+				
+				case IMP_SKILL_TRAITS__CAMO:
+					AddSkillToSkillList( CAMOUFLAGED );
+					break;
+				
+				case IMP_SKILL_TRAITS__MARTIAL_ARTS:
+					AddSkillToSkillList( MARTIALARTS );
+					break;
+			}
```

Comment:
I think coverity will complain about the missing default case. Might be cleaner with a map with SkillTrait, since we can make this file's enum match it completely.